examples/plot_plotting_metrics_with_error.py

```diff
@@ -0,0 +1,252 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+"""
+============================
+Plotting Metrics with Errors
+============================
+"""
+# %%
+# Load and preprocess the data set
+# ================================
+# We start by importing the various modules we're going to use:
+from fairlearn.experimental.enable_metric_frame_plotting import plot_metric_frame
+from fairlearn.metrics import MetricFrame
+from sklearn.datasets import fetch_openml
+from sklearn.metrics import recall_score, accuracy_score, confusion_matrix
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.compose import ColumnTransformer
+from sklearn.impute import SimpleImputer
+from sklearn.preprocessing import StandardScaler, OneHotEncoder
+from sklearn.compose import make_column_selector as selector
+from sklearn.pipeline import Pipeline
+import numpy as np
+
+# %%
+# We download the data set using :meth:`fetch_openml` function in :class:`sklearn.datasets`.
+# The original Adult data set can be found at https://archive.ics.uci.edu/ml/datasets/Adult
+# There are some caveats to using this dataset, but we will use it solely as an example
```

Comment:
I think I added this in response to discussion from a weekly meeting, can you help enumerate what caveats I should include here?

Reply:
@hildeweerts do you have a reference for this?

Reply:
Oof... I believe that sentence was originally based on some random blog post (but it might have just been a conversation with @romanlutz). This paper @LeJit mentioned recently might serve as a reference here: https://arxiv.org/pdf/2108.04884.pdf (but I haven't read it in detail). 

Main caveats IMO are:
    * The 50.000 threshold is arbitrary and does not necessarily correspond to a meaningful ML task.
    * Census data is not collected for making predictions but for describing and studying populations; without further context it is unclear under which circumstances it might be suitable for making predictions.
    * Using a "real-world" data set in a benchmarking fashion ignores all relevant real-world context we need to take into account to actually alleviate fairness-related harms.

So basically it kind of goes against many of the things we try to teach people (abstraction traps, construct validity, etc.)

Reply:
I don't want to throw wrench in the works, but one idea would be to use the health care data set from the SciPy tutorial. In fact in the SciPy tutorial we have a whole section where we calculate and plot the error bars, so we would be able to point to the tutorial for additional context. Thoughts?